### ISSUE TITLE
Kubernetes validation should not fail on CRD's

### ISSUE THREAD

**Issue by pbstriker38** [github/super-linter#891]
Kubeval validation fails when given a K8 configuration with a custom resource definition. The ignore missing schema flag should be used when running `kubeval`.
https://www.kubeval.com/#crds
```
kubeval --ignore-missing-schemas
```

**To Reproduce**
1. Create a file with a CRD
```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: productpage
spec:
  hosts:
  - productpage
  http:
  - route:
    - destination:
        host: productpage
        subset: v1
```

2. Run lint

**Expected behavior**
The file should not be validated since it is an unknown Kubernetes type.

**Screenshots**
![image](https://user-images.githubusercontent.com/5264240/96789809-b3646000-13aa-11eb-9e62-743440f1b3c5.png)

**Comment by ferrarimarco** [comment]
I feel your pain here, but I think this should be configurable, not hard-coded (one way or another).

@admiralAwkbar @GaboFDC @IAmHughes do you think there's room for customized commands in #857?

**Comment by ferrarimarco** [comment]
PS: Kubeval doesn't support configuration files.

**Comment by miro6666** [comment]
> Kubeval validation fails when given a K8 configuration with a custom resource definition. The ignore missing schema flag should be used when running `kubeval`.
> https://www.kubeval.com/#crds
> 
> ```
> kubeval --ignore-missing-schemas
> ```
> 
> **To Reproduce**
> 
> 1. Create a file with a CRD
> 
> ```
> apiVersion: networking.istio.io/v1alpha3
> kind: VirtualService
> metadata:
>   name: productpage
> spec:
>   hosts:
>   - productpage
>   http:
>   - route:
>     - destination:
>         host: productpage
>         subset: v1
> ```
> 
> 1. Run lint
> 
> **Expected behavior**
> The file should not be validated since it is an unknown Kubernetes type.
> 
> **Screenshots**
> If applicable, add screenshots to help explain your problem.
> ![image](https://user-images.githubusercontent.com/5264240/96789809-b3646000-13aa-11eb-9e62-743440f1b3c5.png)

**Comment by ferrarimarco** [comment]
Reopening this one because we abandoned #1667 .

I think this is because we always set the `--strict` option when running kubeval.

@admiralAwkbar what if we make `--strict` as the default value of `KUBERNETES_KUBEVAL_OPTIONS`, so the user can customize that as well?

**Comment by admiralAwkbar** [comment]
@ferrarimarco I think that's likely the best option... Let's get a PR in order :)

**Comment by ferrarimarco** [comment]
Actually, I tried that later. I think you've two issues:

1. Kubeval points to a schema repo that is outdated. You can follow the guidance [here](https://github.com/yannh/kubernetes-json-schema/#kubeval) to point to a more up-to-date schema repository with Kubeval's `--schema-location` option.

1. You're trying to lint an Istio ServiceEntry (I think). There's no schema for ServiceEntry objects neither in the outdated schema repo that Kubeval uses, nor in the one I point out above. You can use Kubeval's `--ignore-missing-schemas` option in this case.

So, to summarize (set this variable in the env section your super-linter action):

```yaml
KUBERNETES_KUBEVAL_OPTIONS: "--ignore-missing-schemas --schema-location https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/"
```

Considering it's not a super-linter issue and there's a workaround, I'm going to close this.
